### DISCUSSION TITLE
I can't make Basis work on my code

### DISCUSSION THREAD

**Discussion by ElMastro** [kinnala/scikit-fem#931] [Q&A] [answered]
Good Afternoon,
Sorry if I post my problem here, but I don't really know where to do anywhere else;
I am trying to build a python script in order to write out the stiffness matrix and the boundaries to use outside scikit-fem;
my code load the mesh:
`try:
	mesh = MeshQuad.load(path)
except:
	print("Error!")`
declare the basis, composed from tetragonal elements
`basis = Basis(mesh, ElementTetP0())`

after this, following the documentation, I tried to assemble the problem with

`form_a = BilinearForm(lambda u, v, w: u * v)
B = asm(unit_load, basis)
A = asm(form_a, basis)`
but if I do
`np.savetxt("/tmp/bar.csv", B, delimiter=",")`
trying to export matrix B with numpy, a vector (1 column) comes out. What shall I do?

Marco

**Comment by kinnala** [comment, marked as answer]
You are assembling B using unit_load which is a predefined LinearForm. LinearForm becomes 1D vector after assembled. Perhaps you mean to save A instead?

**Reply by ElMastro** [reply]
Thank you Dr Gustafsson for the quick answer, 
I think I understand it better, now. So if I would like to calculate it on 3 dimensions, I believe i should use
`B = asm(laplace, basis)` which would assemble the Laplace equation on the mesh. If I am not wrong, I could also take out the boundary vector with `vector_a1 = mesh.boundaries['Boundary_1'];
np.savetxt("v1.csv", vector_a1, delimiter=",");`
My problem is that I found what is u in [Finding degrees-of-freedom](https://scikit-fem.readthedocs.io/en/latest/howto.html), but I haven't understand what w and v are (unfortunately in my university book there isn't a similar convention)
